src/Components/InstancesTable/index.js

```diff
@@ -0,0 +1,88 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { TableComposable, Thead, Tr, Th, Tbody, Td } from '@patternfly/react-table';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons';
+import { ClipboardCopy, Card, Pagination } from '@patternfly/react-core';
+import { useQuery } from 'react-query';
+
+import { SSHUsername } from './helpers';
+import instanceLink from '../Common/InstanceLink';
+import { fetchReservationByProvider } from '../../API';
+
+const PER_PAGE_OPTIONS = [
+  { title: '5', value: 5 },
+  { title: '10', value: 10 },
+  { title: '20', value: 20 },
+  { title: '50', value: 50 },
+];
+const InstancesTable = ({ reservationID, provider, region }) => {
+  const [paginationOptions, setPaginationOptions] = React.useState({ perPage: 5, page: 1, startIdx: 0, endIdx: 5 });
+  const { data: instances } = useQuery(['launchInfo', reservationID], () => fetchReservationByProvider(reservationID, provider), {
+    select: (reservation) => reservation?.instances,
+  });
+
+  const onSetPage = (_, newPage, perPage, startIdx, endIdx) => {
+    setPaginationOptions({ perPage, page: newPage, startIdx, endIdx });
+  };
+
+  const onPerPageSelect = (_, newPerPage, newPage, startIdx, endIdx) => {
+    setPaginationOptions((prevOptions) => ({ ...prevOptions, perPage: newPerPage, page: newPage, startIdx, endIdx }));
+  };
+
+  const instancesPerPage = instances?.slice(paginationOptions.startIdx, paginationOptions.endIdx);
+
+  return (
+    <Card style={{ position: 'relative', marginLeft: '-20%', marginRight: '-20%' }}>
+      <Pagination
+        itemCount={instances?.length || 0}
+        page={paginationOptions.page}
+        onSetPage={onSetPage}
+        perPage={paginationOptions.perPage}
+        onPerPageSelect={onPerPageSelect}
+        perPageOptions={PER_PAGE_OPTIONS}
+        isCompact
+        ouiaId="instances-pagination"
+      />
+      <TableComposable aria-label="instances description table" variant="compact">
```

Comment:
Was thinking of adding OUAID here for the "instance description table" widget.

Reply:
oh, sorry I've not noticed this earlier, we will need to merge as the backend is already merged and it will "break" the current UI, would you mind if I open a follow up bug for this?

Reply:
After merging this I will open the PR to add ouaid.
it's just a one-word change.
Thanks!